### PR TITLE
Make `black` requirement consistent with core Kedro & remove `click` requirement

### PR DESCRIPTION
**Pull request by MerelTheisenQB**
Signed-off-by: Merel Theisen <merel.theisen@quantumblack.com>

## Motivation and Context
The latest click 8.1.0 is incompatible with black versions < 22.3.0

https://github.com/pallets/click/issues/2232
https://github.com/psf/black/issues/2964

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Assigned myself to the PR
- [ ] Added tests to cover my changes

### CODE REVIEW

File: astro-airflow-iris/{{ cookiecutter.repo_name }}/src/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-black==22.1.0
+black>=22.3.0
```
Comment (lorenabalan) [inline astro-airflow-iris/{{ cookiecutter.repo_name }}/src/requirements.txt:1]:
```suggestion
black>=22.3.0, <23.0
```

File: pandas-iris/{{ cookiecutter.repo_name }}/src/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-black==22.1.0
+black>=22.3.0
```
Comment (lorenabalan) [inline pandas-iris/{{ cookiecutter.repo_name }}/src/requirements.txt:1]:
```suggestion
black>=22.3.0, <23.0
```

File: pyspark-iris/{{ cookiecutter.repo_name }}/src/requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-black==22.1.0
+black>=22.3.0
```
Comment (lorenabalan) [inline pyspark-iris/{{ cookiecutter.repo_name }}/src/requirements.txt:1]:
```suggestion
black>=22.3.0, <23.0
```

File: pyspark/{{ cookiecutter.repo_name }}/src/requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-black==22.1.0
+black>=22.3.0
```
Comment (lorenabalan) [inline pyspark/{{ cookiecutter.repo_name }}/src/requirements.txt:1]:
```suggestion
black>=22.3.0, <23.0
```

File: test_requirements.txt [hunk 5]
```diff
@@ -1,7 +1,7 @@
 bandit>=1.6.2, <2.0; python_version > '3.6'
 bandit>=1.6.2, <1.7.2; python_version == '3.6'
 behave>=1.2.6, <2.0
-black==22.1.0
+black>=22.3.0
```
Comment (lorenabalan) [inline test_requirements.txt:4]:
```suggestion
black>=22.3.0, <23.0
```

**Review by lorenabalan — APPROVED** [review]

File: test_requirements.txt [hunk 6]
```diff
@@ -1,7 +1,7 @@
 bandit>=1.6.2, <2.0; python_version > '3.6'
 bandit>=1.6.2, <1.7.2; python_version == '3.6'
 behave>=1.2.6, <2.0
-black>=22.3.0
+black>=22.3.0, <23.0
 click<8.0
```
Comment (noklam) [inline test_requirements.txt:5]: `click<9.0` to match with kedro develop branch

**Review by noklam — CHANGES_REQUESTED** [review]

**Review by lorenabalan — APPROVED** [review]
🏆

File: test_requirements.txt [hunk 7]
```diff
@@ -1,8 +1,7 @@
 bandit>=1.6.2, <2.0; python_version > '3.6'
 bandit>=1.6.2, <1.7.2; python_version == '3.6'
 behave>=1.2.6, <2.0
-black==22.1.0
-click<8.0
+black~=22.0
```
Comment (lorenabalan) [inline test_requirements.txt:4]: We can also remove L2 and the python version thing on L1.

**Review by noklam — APPROVED** [review]
I see the CI is failing for `kedro-telemetry`, do we also have py39 py310 support for the plugins or it will comes after `0.18` release?

**Comment by MerelTheisenQB** [comment]
> I see the CI is failing for `kedro-telemetry`, do we also have py39 py310 support for the plugins or it will comes after `0.18` release?

We'll need to add that support for all plugins after kedro `0.18.0` is released.

**Pull request merged**